Modules/mathmodule.c

```diff
@@ -2163,6 +2163,27 @@ math_ldexp_impl(PyObject *module, double x, PyObject *i)
     } else {
         errno = 0;
         r = ldexp(x, (int)exp);
+#ifdef _MSC_VER
+        if (-DBL_MIN < r && r < DBL_MIN) {
```

Comment:
I can't prove this, but my intuition is that denorms are very rare in binary64 work, and that when they may occur they're usually positive. If so it could pay a bit to swap the order of those tests: `r < DBL_MIN` will usually fail regardless, but even if there is a denorm I expect it to fail more often than `-DBL_MIN < r` (which a positive r always passes).

Take it or leave it, as you like 😄.